### ISSUE TITLE
--vars for dbt_project.yml configs: CLI works, RPC doesn't

### ISSUE THREAD

**Issue by jtcohen6** [fishtown-analytics/dbt#3126]
In #2265, we found that when a `--vars` value changed information in the parsed manifest, the RPC server needed to reload/rewrite the manifest. This wasn't an issue with partial parsing—the files were correctly invalidated—but still the server took the manifest parsed by `dbt rpc` as its starting point, when `--vars` can override key configurations in that manifest. The fix for this, in #2363, impacted the performance of the `cli_args` RPC method accordingly.

It seems that the "reload" is still not working for **dbt_project.yml**, however, or at least for configs that contain `{{ var() }}` calls. To be honest, usage of `vars` within the project file is tricky because the parsing/rendering order is not all that clear (#3105). We should do core work here to rationalize and clarify when `vars` are set, and when they're available (#2955).

In the meantime, we should figure out why RPC behavior is not consistent with CLI behavior.

### Steps To Reproduce

In **models/my_model.sql**:
```sql
select '{{ var("my_var", "foo") }}'
```

In **dbt_project.yml**:
```yml
models:
  my_project:
    +schema: "{{ var('my_var', 'foo') }}"
```

#### CLI
```
$ dbt run --vars '{foo: bar}'
```
```
  create view "jerco"."dbt_jcohen_bar"."my_model__dbt_tmp" as (
    select 'bar'
  );
```

#### RPC

Request:
```
{
    "jsonrpc": "2.0",
    "method": "cli_args",
    "id": "2db9a2fe-9a39-41ef-828c-25e04dd6b07d",
    "params": {
        "timeout": 60,
        "cli": "run --vars '{my_var: bar}'"
    }
}
```
Poll for result:
```json
{
    "result": {
        ...
        "logs": [
            ...
            {
                "message": "On model.my_project.my_model: /* {\"app\": \"dbt\", \"dbt_version\": \"0.19.0\", \"profile_name\": \"garage-postgres\", \"target_name\": \"dev\", \"node_id\": \"model.my_project.my_model\"} */\n\n  create view \"jerco\".\"dbt_jcohen_foo\".\"my_model__dbt_tmp\" as (\n    select 'bar'\n  );\n",
                ...
            },
            ...
        ],
        "results": [
            {
                "node": {
                    "schema": "dbt_jcohen_foo",
                    "compiled_sql": "select 'bar'",
                    "relation_name": "\"jerco\".\"dbt_jcohen_foo\".\"my_model\"",
                    ...
                },
                ...
            },
            ...
        ]
    }
}
```

Of course, there's one impractical way to get this to work in the meantime:
```
$ dbt rpc --vars '{my_var: bar}'
```
Then a simple `run` method returns the expected result.

**The output of `dbt --version`:**
```
$ dbt --version
installed version: 0.19.0
   latest version: 0.19.0

Up to date!
```

**Comment by misurin** [comment]
--vars doesn't seem to work with built-in project configs either. I pass `--vars 'log-path: ["/tmp/dbt_logs"]'` to my `dbt run` command and  it still writes logs to the project dir.

**Comment by jtcohen6** [comment]
Yes, `--vars` passed to `cli_args` RPC methods do not properly update expressions that use `{{var()}}` anywhere in `dbt_project.yml` today.

(It's also not _exactly_ documented that `--vars` will work for a path configs like `log-path`, which (to my mind) makes more sense as an environment variable. We do document the availability of `--vars` in the context of resource configs [here](https://docs.getdbt.com/reference/dbt-jinja-functions/dbt-project-yml-context). I don't think we intend to change that behavior, but it's worth saying again that we need to rationalize and clarify when exactly vars are available.)

**Comment by kconvey** [comment]
Any idea when this might get resolved?
